org.alloytools.alloy.core/src/main/java/edu/mit/csail/sdg/parser/CompModule.java

```diff
@@ -85,12 +85,15 @@
 import edu.mit.csail.sdg.ast.Sig.PrimSig;
 import edu.mit.csail.sdg.ast.Sig.SubsetSig;
 import edu.mit.csail.sdg.ast.Type;
+import edu.mit.csail.sdg.ast.VisitQuery;
 import edu.mit.csail.sdg.ast.VisitQueryOnce;
 import edu.mit.csail.sdg.ast.VisitReturn;
 
 /**
  * Mutable; this class represents an Alloy module; equals() uses object
  * identity.
+ *
+ * @modified: Nuno Macedo // [HASLab] electrum-temporal
```

Comment:
other than adapting the methods to support commands with trace prefix length scopes, changes here are for the detection of errors and warnings during signature/field/command resolution; note that up to now signatures did not throw warnings since those arose mainly from type system warnings;

the following warnings are thrown:
- if a static sig is `in` a variable sig (since this makes part of the parent immutable)
- if a static sig `extends` a variable sig (since this makes part of the parent immutable)
- if a variable sig `extends` a static sig (redundant since it will be immutable anyway)
- if a static field is declared in a variable sig (since this makes part of the sig immutable)
- if a static field is declared with a variable bounding expression (since this makes part bounding expression immutable)

the following errors are thrown:
- if scopes are assigned to non-top-level variable sigs
- if exact scopes are assigned to variable sigs